### PR TITLE
[quant][core] Added support for quantizing a constant value and broadcasting the assignment to an entire tensor

### PR DESCRIPTION
**Pull request by dzdang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78562
* #78662
* #78661

Summary:
Support in QuantizedOpKernels.cpp exists for quantizing a
fp tensor and copying the quantized tensor to the output (quantized)
tensor. This PR adds analogous support for quantizing a scalar input and
broadcasting the quantized value to the entire tensor.

This PR is intended to be used for fill_quantized, where, currently, a
tensor filled with the same quantized value needs to be created in order
to be assigned to an output quantized tensor, resulting in an
inefficient fill operation.

Test plan:
See the subsequent PR as this PR does not make changes that would affect
front end code.

Differential Revision: [D36816604](https://our.internmc.facebook.com/intern/diff/D36816604)

### CODE REVIEW

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36816604).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36816604).

File: aten/src/ATen/native/quantized/affine_quantizer.h [hunk 1]
```diff
@@ -7,18 +7,39 @@
 namespace at {
 namespace native {
 
+Tensor& quantize_scalar_per_tensor_affine(
+    const float val,
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/affine_quantizer.h:11]: why is this a float? should we be using Scalar to support different types of Scalars?

**Comment by jerryzh168** [comment]
looks like we duplicated a lot of code here, have we considered just construct a Tensor from the scalar value and reuse the old code?

**Comment by dzdang** [comment]
> looks like we duplicated a lot of code here, have we considered just construct a Tensor from the scalar value and reuse the old code?

@jerryzh168  yes I did. the problem is the existing functions in fbgemm expects 2 inputs tensors to have the same storage size, so it won't work with a single element tensor. I would have to modify the fbgemm backend implementation if we want to make those functions compatible and I think we previously discussed that we should just overload it in pytorch?

**Pull request closed without merging**